File: java/src/leetcode/P0344.java
package leetcode;

public class P0344 {
    public static void main(String[] args) {
        char[] s = {'H','a','n','n','a','h'};
//        char[] s = {'H', 'e', 'l', 'l', 'o'};
        reverseString(s);
        System.out.println(s);
    }

    public static void reverseString(char[] s) {
        char temp = ' ';
        for (int i = 0; i < s.length/2; i++) {
            temp = s[i];
            s[i] = s[s.length - 1 - i];
            s[s.length - 1 - i] = temp;
        }
    }
}

File: ruby/leetcode/problems/P0344.rb
# 344. Reverse String

# @param {Character[]} s
# @return {Void} Do not return anything, modify s in-place instead.
def reverse_string(s)
  # s.reverse
  left = 0
  right = s.size - 1
  while left < right
    temp = ''
    temp = s[left]
    s[left] = s[right]
    s[right] = temp
    left += 1
    right -= 1
  end
  s
end
s = ["h","e","l","l","o"]
puts reverse_string(s)